libs/common/src/test/java/org/opensearch/common/round/RoundableTests.java

```diff
@@ -10,17 +10,34 @@
 
 import org.opensearch.test.OpenSearchTestCase;
 
+import java.util.List;
+
 public class RoundableTests extends OpenSearchTestCase {
 
-    public void testFloor() {
-        int size = randomIntBetween(1, 256);
-        long[] values = new long[size];
-        for (int i = 1; i < values.length; i++) {
-            values[i] = values[i - 1] + (randomNonNegativeLong() % 200) + 1;
-        }
+    public void testRoundingEmptyArray() {
+        Throwable throwable = assertThrows(IllegalArgumentException.class, () -> RoundableFactory.create(new long[0], 0));
+        assertEquals("at least one value must be present", throwable.getMessage());
+    }
+
+    public void testRoundingSmallArray() {
+        int size = randomIntBetween(1, 64);
+        long[] values = randomArrayOfSortedValues(size);
+        Roundable roundable = RoundableFactory.create(values, size);
+
+        assertEquals("BidirectionalLinearSearcher", roundable.getClass().getSimpleName());
+        assertRounding(roundable, values, size);
+    }
 
-        Roundable[] impls = { new BinarySearcher(values, size), new BidirectionalLinearSearcher(values, size) };
+    public void testRoundingLargeArray() {
+        int size = randomIntBetween(65, 256);
+        long[] values = randomArrayOfSortedValues(size);
+        Roundable roundable = RoundableFactory.create(values, size);
 
+        assertTrue(List.of("BtreeSearcher", "BinarySearcher").contains(roundable.getClass().getSimpleName()));
```

Comment:
is it possible to validate that the factory is returning BinarySearcher when the code is built using JDK20+?

Reply:
I previously had assertions for this, but it required reflective lookup of `IS_BTREE_SEARCH_SUPPORTED` as this field won't be available below Java 20. Since this is an implementation detail, the suggestion was to assert on things that are exposed.

Check out this thread: https://github.com/opensearch-project/OpenSearch/pull/11194#discussion_r1416161203

Reply:
Sorry! didn't realize it was already discussed. Makes sense! Thanks!